Requirements/cross-chain-transfer.md

```diff
@@ -68,35 +70,43 @@ A leader that is now synced and observing unprocessed previous transfers will st
+**The structure of the outputs of the transaction:**
+- Outputs, if more than one is required, are sorted by destination address (i.e. the scriptPubKey).
+- The output returning change back to the multisig, if any, appears after the value transfer output(s).
+- An OP_RETURN output containing the hash of the block on the originating chain (i.e. the block the deposit transactions appear in).
+
+A member that comes back online after a long period will need to sync both chains and also build the database of transfer transactions, they should be able to do this simply by looking at the chains themselves.  
+A member that comes back online will, once it is done syncing both its chains and has the correct views of the multisig UTXO lists, check all the unprocessed transfers, start building and signing corresponding transactions, and broadcast them to its peers. 
 
 ![Building transactions](../assets/cross-chain-transfers/building-transaction.svg)
 
-**The algorithem to select the leader:**
+**The algorithm to select the leader:**
 
-TODO:
-either reuse the current mechanism : hash block hash with public keys of members and order them alphabetically to get a leader table
-or have the leader take turn in a pre-established way based on height of block modulo federation size (somewhat similar to POA)
+The leader role will be assigned round-robin to each member in turn, following the order they are defined in the setup. It is not considered to be necessary to introduce the complexity of a pseudorandom-assignment methodology such as the block hash.
 
 ## Considerations ## 
 
-- What is the fee going to be on the transfer trx (to maintain determinism)
-- What if a block is full of transfer transactions (too full for the other chain)
-- What if all members are offline how do we handle resync of federation members
-
-What can cause a member to generate a multisig that is different for the rest of the federation (break determinism)
+- What is the fee going to be on the transfer trx (to maintain determinism)?
+- What if a block is full of transfer transactions (too full for the other chain)?
+- What if all members are offline how do we handle resync of federation members?
+- If a member comes online as the current leader, without having been present for a previous unprocessed block, they will need to get signatures from the other members for that previous block. This implies the need for some form of rebroadcast mechanism, presumably initiated by the current leader.
+- If the maturity of a UTXO has to be considered (i.e. due to the possibility of reorgs), then there is a bootstrapping issue when the network is first initialised. The premine is a single UTXO, and would have to be split in order for the chain to have adequate UTXO bandwidth. This splitting would have to be done on a continuous basis to preserve a pool of available UTXOs of appropriate sizes. These sizes are difficult to anticipate in advance as they would depend on the particular sidechain.
```

Comment:
Nice one, we thought about splitting the 20 millions in the very first blocks, but not about maintaining a healthy supply across time! It might add a bit of complexity :/ we probably need to see if that fits in this first prod version...